include/internal/fips.h

```diff
@@ -16,6 +16,39 @@
 /* Return 1 if the FIPS self tests are running and 0 otherwise */
 int ossl_fips_self_testing(void);
 
+/* Deferred KAT tests categories */
+
+/*
+ * The Integrity category is used to run test that are required by the
+ * integrity check and are a special category that can therefore never
+ * really be deferred. Keep it commented here as a reminder.
+ * #  define FIPS_DEFERRED_KAT_INTEGRITY 0
+ */
+#  define FIPS_DEFERRED_KAT_CIPHER 1
+#  define FIPS_DEFERRED_KAT_ASYM_CIPHER 2
+#  define FIPS_DEFERRED_KAT_ASYM_KEYGEN 3
+#  define FIPS_DEFERRED_KAT_KEM 4
+#  define FIPS_DEFERRED_KAT_DIGEST 5
+#  define FIPS_DEFERRED_KAT_SIGNATURE 6
+#  define FIPS_DEFERRED_KAT_KDF 7
+#  define FIPS_DEFERRED_KAT_KA 8
+#  define FIPS_DEFERRED_MAX 9
```

Comment:
MAC & RAND are missing.
MAC aren't tested independently but RANDs (drbgs) are.

Reply:
I was more thinking of someone who uses CMAC but not the DRBG.
Currently, the only CMAC test we have is the DRBG one which is lots more expensive than just testing CMAC.  Having the CMAC trigger this is possible but it isn't self contained.

However, we don't want to add an individual CMAC test that runs unconditionally since the DRBG can do it for us.  Your _also satisfies_ would deal with this case I think.